protocols/mdns/src/behaviour/iface/query.rs

```diff
@@ -247,21 +244,16 @@ impl MdnsPeer {
             .flat_map(|txt| txt.iter())
             .filter_map(|txt| {
                 // TODO: wrong, txt can be multiple character strings
-                let addr = match dns::decode_character_string(txt) {
-                    Ok(a) => a,
-                    Err(_) => return None,
-                };
+                let addr = dns::decode_character_string(txt).ok()?;
+                
                 if !addr.starts_with(b"dnsaddr=") {
                     return None;
                 }
-                let addr = match str::from_utf8(&addr[8..]) {
-                    Ok(a) => a,
-                    Err(_) => return None,
-                };
-                let mut addr = match addr.parse::<Multiaddr>() {
-                    Ok(a) => a,
-                    Err(_) => return None,
-                };
+
+                let addr = str::from_utf8(&addr[8..]).ok()?;
+
+                let mut addr = addr.parse::<Multiaddr>().ok()?;
```

Comment:
We could even chain these together here :)

```suggestion
                let mut addr = str::from_utf8(&addr[8..]).ok()?.parse::<Multiaddr>().ok()?
```